tests/framework/extensions/machinepools/machinepools.go

```diff
@@ -53,7 +55,25 @@ type NodeRoles struct {
 	Quantity     int32 `json:"quantity" yaml:"quantity"`
 }
 
-// RKEMachinePoolSetup is a helper method that will loop and setup muliple node pools with the defined node roles from the `nodeRoles` parameter
+func (n NodeRoles) String() string {
+	result := make([]string, 0, 3)
+	if n.Quantity < 1 {
+		return ""
+	}
+	if n.ControlPlane {
+		result = append(result, "controlplane")
+	}
+	if n.Etcd {
+		result = append(result, "etcd")
+	}
+	if n.Worker {
+		result = append(result, "worker")
+	}
+
+	return fmt.Sprintf("%d %s", n.Quantity, strings.Join(result, "+"))
+}
+
+// RKEMachinePoolSetup is a helper method that will loop and setup multiple node pools with the defined node roles from the `nodeRoles` parameter
```

Comment:
Looks like lines 76-92 are duplicate code. Why is it commented out? Can it be removed?

Reply:
It's a comment explaining how it works. Although the comment does seem out of date, I'll update.

Reply:
I have, as a side note I like the `machinepools.NodeRools` mechanism it's really easy to work with and it's really easy to structure the json for in `CATTLE_TEST_CONFIG`.